misc/connection-limits/src/lib.rs

```diff
@@ -466,6 +466,57 @@ mod tests {
         quickcheck(prop as fn(_));
     }
 
+    #[test]
+    fn max_established_incoming_downcast_ref() {
```

Comment:
I don't think we need this test. We don't actually control any logic here other than forwarding calls so the typesystem should be good enough :)

Reply:
Should I remove the test or just leave it be?